launcher/src/main/java/org/apache/spark/launcher/PySparkLauncher.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.launcher;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Launcher for PySpark.
+ * <p/>
+ * Handles parsing command line options passed to the pyspark script. This allows
+ * sharing that logic with other launchers, to keep them in sync.
+ */
+class PySparkLauncher extends AbstractLauncher<PySparkLauncher> {
+
+  private final List<String> args;
+
+  PySparkLauncher(List<String> args) {
+    this.args = args;
+  }
+
+  @Override
+  protected List<String> buildLauncherCommand() throws IOException {
+    SparkSubmitCliLauncher launcher = new SparkSubmitCliLauncher(true, args);
```

Comment:
Ok, by "going through spark-submit" I mean it calls spark-submit, directly or indirectly. Previously it would still call `bin/spark-submit` as a subprocess in `java_gateway.py`.